hub/core/storage/lru_cache.py

```diff
@@ -270,3 +270,31 @@ def _list_keys(self):
         for key in self.cache_storage:
             all_keys.add(key)
         return list(all_keys)
+
+    def __getstate__(self):
+        """Returns the state of the cache, for pickling"""
+
+        # flushes the cache before pickling
+        if not hasattr(self, "read_only") or not self.read_only:
+            self.flush()
```

Comment:
i've seen this block multiple times throughout the code, can you move to a util or make it a property of `StorageProvider` base class?

if you do this, could you also please change all other places where this block is used to the helper method?

Reply:
something like `self.try_flush()` or `self.flush_if_not_read_only()`